addons/stock/models/stock_quant.py

```diff
@@ -147,14 +147,17 @@ def create(self, vals):
         """ Override to handle the "inventory mode" and create a quant as
         superuser the conditions are met.
         """
+        product = self.env['product.product'].browse(vals['product_id'])
```

Comment:
Why adding this. The constraint will do the job, no ?

Reply:
This was a safety thing in case a `_update_available_quantity` is called on a consumable without a package (also to keep the code cleaner without extra checks)

Reply:
Yeah I understand with your other change, my bad